modin/pandas/test/utils.py

```diff
@@ -584,7 +590,7 @@ def check_df_columns_have_nans(df, cols):
     """
     return (
         pandas.api.types.is_list_like(cols)
-        and any(x in df.columns and df[x].hasnans for x in cols)
+        and any(isinstance(x, str) and x in df.columns and df[x].hasnans for x in cols)
```

Comment:
if `x` is `Series` shouldn't we check that its `name in df.columns` and if it is, check if that column contains nans?

Reply:
Hm. If `name` is in df.columns and the `_parent` is the df, probably, yes.